src/components/SearchBar.tsx

```diff
@@ -499,4 +509,17 @@ export function calculateEntityPreviewsCount(children: ReactNode): number {
     return c;
   });
   return count;
+}
+
+/**
+ * Sets screenreader text for navigable entity previews from a ReactNode containing DropdownItems.
+ */
+export function setPreviewScreenreader(children: ReactNode): ReactNode {
```

Comment:
Is there no other way to set the ariaLabel? We want to avoid low level methods like cloneElement and our recursivelyMapChildren (ideally we never have to use recurisvelymapChildren, there's also some issues with it currently)

Reply:
With the way the `VisualAutocompleteConfig` is set up, the developer has to pass in their own `renderEntityPreviews` function and Nidhi and I decided we didn't want to leave it up to them to add an aria-label, so I go back and dynamically change the prop after the EntityPreviews have been rendered. Do you have any ideas on how to get around that?

Reply:
It might also be worth checking with product on whether the developer should be in charge of this or if we should add it "magically"

Reply:
I was thinking we could update the onSubmit param in renderEntityPreview to instead be an object with default props people could use.

```tsx
{
  renderEntityPreviews: (isLoading, verticalKeyToResults, props) => {
    return verticalKeyToResults.people.map((r, index) => <DropdownItem {...props}>blah</DropdownItem>)
  }
}
```

Reply:
We could also add a default value to DropdownItem, though I didn't like that since that's putting SearchBar details into the Dropdown

Reply:
both recent searches and autocomplete options have the "magical" screenreader text set up so I think what they wanted was for it to be consistent with entity previews as well. I can ask though!

Reply:
This is similar to a pattern react-bootstrap/restart-ui use, where they export a hook like `useDropdownItem` that provides props that people can use in whatever container element they want https://github.com/react-bootstrap/react-bootstrap/blob/af953c20361d91a50c06b75c8bc7d6bb13da3165/src/DropdownItem.tsx#L5

Reply:
oh I only just saw the other comments, but I imagine they will like things to be magical by default as well!

Reply:
The default aria-label should probably just be the value of the item anyways; I'll add that in

Reply:
I think, by default, the screenreader will read the value without a prefix even if no aria-label is specified